File: app/src/main/java/com/github/Doomsdayrs/apps/shosetsu/recycleObjects/NovelCard.java
package com.github.Doomsdayrs.apps.shosetsu.recycleObjects;

import java.net.URI;

public class NovelCard {
    public String libraryImageResource;
    public String title;
    public String URL;

    public NovelCard(String libraryImageResource, String title,URI URL){
        this.libraryImageResource = libraryImageResource;
        this.title = title;
    }

    public NovelCard(String title, URI URL){
        this.libraryImageResource = null;
        this.title = title;
    }


}

File: app/src/main/java/com/github/doomsdayrs/apps/shosetsu/variables/variables/recycleObjects/NovelCard.kt
package com.github.doomsdayrs.apps.shosetsu.variables.recycleObjects

/*
 * This file is part of Shosetsu.
 *
 * Shosetsu is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Shosetsu is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Shosetsu.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================
 */
/**
 * Shosetsu
 * 9 / June / 2019
 *
 * @author github.com/doomsdayrs
 * <p>
 *     Database representation of a novel and it's information
 * </p>
 */
class NovelCard(title: String, val novelID: Int, val novelURL: String, val imageURL: String, val formatterID: Int) : RecycleCard(title)